planetiler-core/src/main/java/com/onthegomap/planetiler/util/LoopLineMerger.java

```diff
@@ -185,12 +219,37 @@ public List<LineString> getMergedLineStrings() {
     }
 
     if (minLength > 0.0) {
-      removeShortStubEdges();
-      merge();
-      removeShortEdges();
+      double step = 1.0 / precisionModel.getScale();
+      for (double stubMinLength = 0.0; stubMinLength < minLength; stubMinLength += step) {
+        removeShortStubEdges(stubMinLength);
+        merge();
+      }
+      removeShortStubEdges(minLength);
       merge();
+      // minLength = 10 * 0.0625;
+      // removeShortEdges();
+      // merge();
```

Comment:
I commented out regular edge dropping based on length, because after stub removal, we only have fully connected edges left here. And we want to keep these for angle merging after simplification...